pages/web-scraping/index.jsx

```diff
@@ -99,40 +82,40 @@ export const WebScrapingUseCase = ({ from, to, disableBotDetection }) => {
     { immediate: false }
   );
 
-  async function searchFlights() {
-    setLoading(true);
-    setMessage('');
-    const { requestId } = await getVisitorData();
-    try {
-      const response = await fetch(`/api/web-scraping/flights`, {
-        method: 'POST',
-        headers: {
-          'Content-Type': 'application/json',
-        },
-        body: JSON.stringify({
-          from: fromCode,
-          to: toCode,
-          requestId,
-        }),
-      });
-      if (response.status === 500) {
-        throw new Error(response.statusText);
-      }
-      /** @type {import('../../server/checkResult').CheckResult} */
-      const result = await response.json();
-      setLoading(false);
-      setFlights(result.data);
-      if (result.severity !== 'success') {
-        setMessage(result.message);
-        setMessageSeverity(result.severity);
-      }
-    } catch (error) {
-      setLoading(false);
-      setMessageSeverity('error');
-      setMessage(error.toString());
-      console.log(error);
+  const getFlightsQuery = useQuery(
+    ['getFlights'],
+    async () => {
+      const { requestId } = await getVisitorData();
+      return await (
+        await fetch(`/api/web-scraping/flights`, {
+          method: 'POST',
+          headers: {
+            'Content-Type': 'application/json',
+          },
+          body: JSON.stringify({
+            from: fromCode,
+            to: toCode,
+            requestId,
+          }),
+        })
+      ).json();
+    },
+    {
+      enabled: false,
+    }
+  );
+
+  // Search for flights when the page loads
+  useEffect(() => {
+    if (fromCode && toCode) {
+      getFlightsQuery.refetch();
     }
-  }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  const { isFetching, data } = getFlightsQuery;
+  const flights = data?.data;
+  console.log(isFetching);
```

Comment:
Remember to remove the console.log :)

Reply:
Hmm I don't see it there, maybe you commented on an old commit?

Reply:
Probably yeah, my mistake!